### DISCUSSION TITLE
UWP backend for .NET MAUI.

### DISCUSSION THREAD

**Discussion by Qws** [dotnet/maui#15613] [Ideas]
### Description

Xamarin for Windows is powered by UWP. React Native is also powered by UWP.

But MAUI is powered by... WinUI3.

For me, WinUI3 is just not good enough. It has some major issues. We are talking about old WinUI3 issues, still not fixed. Perhaps never will be fixed.

1. WinUI3 apps have slower startup time. Microsoft straight up stated that WinUI3 will never have UWP's startup speed.
2. No splashscreen. So you may click multiple times to try and start the app, believing that the app is unresponsive. There is no loading feedback.
3. Darkmode WinUI3 apps have white flashing on startup. Because every Win32 app starts with white background. This is very annoying.

The WinUI3 issues are for me dealbreakers... I'm not even talking about app-size or sandboxing.

But the app startup is very important. An app needs to be quick and responsive, that's the first impression user get from the app.

Xamarin en React Native are using UWP. Because Xamarin is ending, I think it's good idea for MAUI to continue using UWP for people that need to create quality UX apps.

### Public API Changes

Unlike WPF which has older XAML dialect, WinUI3 and UWP have similar XAML/API syntax. So it shouldn't be that difficult to map and maintain both of them?

React Native also supports UWP and WinUI3 simultaneously.

### Intended Use-Case

UWP has its own strong advantages, like .NET Native (high performance) and security (sandboxed). If you want to push faster, lighter MAUI apps, give us the option to use UWP instead of WinUI3.

**Comment by janseris** [comment]
> Darkmode WinUI3 apps have white flashing on startup. Because every Win32 app starts with white background. This is very annoying.

What about delay showing the main window until properly rendered?

**Comment by milklike** [comment]
WinUI3's performance is really bad.But you could not change their decision，because they don't want to discuss this issue head-on.

**Comment by enisn** [comment]
I had some other problems with WinUI3. Does anyone know if there is any UWP adapter for MAUI?

**Comment by baaaaif** [comment]
I think Microsoft has been listening and Maui AND WinUI3 have received a lot of stability and performance-enhancing fixes recently.

Give
```
<ItemGroup Condition="'$(TargetFramework)' == '$(NetVersion)-windows10.0.19041.0'">
   <PackageReference Include="Microsoft.WindowsAppSDK" Version="1.4.240211001" />
</ItemGroup>
```
a chance. It is fast and stable.
Things are working great if you invest time in optimizing your xaml and your ViewModel. p.ex. wrong or correctly used async/await  can be a game changer
